model/riscv_extensions.sail

```diff
@@ -0,0 +1,61 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+scattered enum extension
+
+// Bit manipulation
+enum clause extension = Ext_B
+// Compressed
+enum clause extension = Ext_C
+// Double precision float
+enum clause extension = Ext_D
+// Single precision float
+enum clause extension = Ext_F
+// Multiply/Divide; not Machine!
+enum clause extension = Ext_M
+// Supervisor
+enum clause extension = Ext_S
+// Supervisor time compare
+enum clause extension = Ext_Sstc
```

Comment:
Other than `Sstc`, only single-letter extensions have comments. This should probably be consistent. 

Though to be honest, I think it makes more sense to either have comments for all the extensions or none of them. There is nothing special about single-letter ones in this regard.

Reply:
I was mainly being lazy, but I have now tediously gone through the spec and copied the official extension names into comments for every one (where possible; Zmmul and Zca didn't have explicit names).

Reply:
I guess this could be a nice `mapping extensionName` at some point but lets do that later.